src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -233,6 +233,11 @@ public partial class EventSource : IDisposable
 #pragma warning restore CA1823
 #endif //FEATURE_EVENTSOURCE_XPLAT
 
+        private static bool IsSupported { get; } = InitializeIsSupported();
```

Comment:
How does the linker do the stub replacement for a member like this?  This construct will be compiled down to something like:
```C#
private static bool <>_isSupported;

static /*beforefieldinit*/ EventSource
{
    <>_isSupported = InitializeIsSupported();
}

private static bool IsSupported => <>_isSupported;
```
When the System.Diagnostics.Tracing.EventSource.IsSupported linker feature is set, is the linker able to unwind all of that back to just:
```C#
private static bool IsSupported => false;
```
or do we end up with:
```C#
private static bool <>_isSupported;

static /*beforefieldinit*/ EventSource
{
    <>_isSupported = InitializeIsSupported();
}

private static bool IsSupported => false;
```
?

Reply:
You get the latter:

```C#
        static EventSource()
        {
            EventSource.<IsSupported>k__BackingField = EventSource.InitializeIsSupported();
        }

        private static bool IsSupported
        {
            get
            {
                return false;
            }
        }
```

We discussed this a bit here: https://github.com/mono/linker/issues/1112#issuecomment-627506851. Do you think this will be an issue leaving this dead field and method?

cc @vitek-karas @MichalStrehovsky

Reply:
I wouldn't block on it; we're only going to have a relatively small number of switches, and if each ends up leaving behind this small overhead, so be it.  It'd just be nice if the linker could eventually be improved to do a better job with it.